### PR TITLE
Wallet: Bug fixes for Onboarding

### PR DESCRIPTION
**Pull request by figitaki**
Cleans up some UX feedback on the custom and express onboarding paths.

- Removes the back button once user reaches "Starting your node"
- Adds an error state to the Installing Coda section
- Prevents duplicate express installation
- Fixes the MacOS ci

### CODE REVIEW

**Comment by bkase** [comment]
Nix is going to be a pain to get working on the newer macOS -- do we need to upgrade the Xcode version to get the rest of the build to work?

**Comment by bkase** [comment]
I'm trying with an older xcode and seeing if that will pass

**Comment by bkase** [comment]
Prod builds of the wallet are failing:

```
    TypeError: Cannot read property 'getPath' of undefined

      18 | var Command = { };
      19 | 
    > 20 | var installPath = Electron.app.getPath("userData") + "/daemon";
         |                                ^
      21 | 
      22 | function codaCommand(port, extraArgs) {
      23 |   var env = Process.env;
```

**Comment by bkase** [comment]
There's a problem with the keychain loading on fastlane

https://app.circleci.com/pipelines/github/CodaProtocol/coda/19168/workflows/f15542e4-ec9a-480b-971d-30a923ef2337/jobs/359655

File: frontend/wallet/src/render/CodaProcess.re [hunk 1]
```diff
@@ -59,17 +59,20 @@ let useHook = () => {
...
   dispatch;
 };
+
+let useStartEffect = dispatch => {
+  React.useEffect0(() => {
+    let token = MainCommunication.listen();
+    let args =
+      switch (getLocalStorageNetwork()) {
+      | "" => []
+      | s => ["-peer", s]
```
Comment (Schmavery) [inline frontend/wallet/src/render/CodaProcess.re:71]: I realize this is just uncommented but the behavior here with getLocalStorageNetwork, defaultNetwork and defaultPeers seems a little funny/wrong

File: frontend/wallet/src/render/DownloadLogic.re [hunk 2]
```diff
@@ -52,8 +53,12 @@ let handleResponse = (response, filename, dataEncoding, chunkCb, doneCb) => {
     |> int_of_string;
   let stream =
     Bindings.Stream.Writable.create(filename, {"encoding": dataEncoding})
-    -> Bindings.Stream.Writable.onFinish(() => doneCb(Belt.Result.Ok()))
-    -> Bindings.Stream.Writable.onError(e => doneCb(Error(e)));
+    ->Bindings.Stream.Writable.onFinish(() => doneCb(Belt.Result.Ok()))
+    ->Bindings.Stream.Writable.onError(e =>
+        doneCb(
+          Error(Tc.Option.withDefault(~default="", Js.Exn.message(e))),
```
Comment (Schmavery) [inline frontend/wallet/src/render/DownloadLogic.re:59]: Should we have a default error message here

File: frontend/wallet/src/render/views/onboarding/InstallCodaStep.re [hunk 3]
```diff
@@ -46,31 +44,48 @@ module InstallProgress = {
...
+             <p className=Styles.downloaderSubtext> {React.string(err)} </p>
+           </>
+         | _ =>
+           <>
+             <p className=Styles.installingText>
+               {React.string("Installing Coda")}
+             </p>
+             <a
+               onClick={_ =>
+                 openExternal("https://codaprotocol.com/docs/troubleshooting")
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/onboarding/InstallCodaStep.re:81]: fwiw I don't think this page has much that would help with this right now but that doesn't need to be in this PR

File: frontend/wallet/src/render/views/onboarding/Onboarding.re [hunk 4]
```diff
@@ -38,37 +38,38 @@ let make = () => {
     | CustomSetupB =>
       <CustomSetupB
         prevStep={_ => setOnboardingStep(_ => CustomSetupA)}
-        runNode={_ => setOnboardingStep(_ => RunNode)}
+        runNode={_ => setOnboardingStep(_ => RunNode(false))}
         nextStep={_ => setOnboardingStep(_ => CustomSetupC)}
       />
     | CustomSetupC =>
       <CustomSetupC
         prevStep={_ => setOnboardingStep(_ => CustomSetupB)}
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/onboarding/Onboarding.re:46]: Does anyone know what the deal with CustomSetupB is? @michellewong793 
It's never the nextStep but here it's the prevStep

File: frontend/wallet/src/render/views/onboarding/RunNodeStep.re [hunk 5]
```diff
@@ -29,7 +30,11 @@ module Styles = {
 };
 
 [@react.component]
-let make = (~prevStep, ~createAccount) => {
+let make = (~prevStep as _, ~createAccount, ~managed) => {
+  // Start the daemon process if we're in managed setup
+  let dispatch = React.useContext(ProcessDispatchProvider.context);
+  managed ? CodaProcess.useStartEffect(dispatch) : ();
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/onboarding/RunNodeStep.re:36]: Calling a hook in a conditional https://reactjs.org/docs/hooks-rules.html

File: frontend/wallet/src/render/views/transactions/Transactions.re [hunk 6]
```diff
@@ -172,6 +172,7 @@ let make = () => {
        let transactionQuery =
          TransactionsQueryString.make(
            ~publicKey=Apollo.Encoders.publicKey(pubkey),
+           ~after="",
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/transactions/Transactions.re:175]: is this required?

**Comment by bkase** [comment]
CI passes 🎉

File: frontend/wallet/src/render/views/transactions/Transactions.re (hunk 6 above)
Comment (figitaki) [reply frontend/wallet/src/render/views/transactions/Transactions.re:175]: Yes, this fixes a bug in the Transactions view. Maybe related to upgrading bs-platform?

File: frontend/wallet/src/render/views/onboarding/Onboarding.re (hunk 4 above)
Comment (figitaki) [reply frontend/wallet/src/render/views/onboarding/Onboarding.re:46]: We decided to move this out if I recall our conversation with Chris.

File: frontend/wallet/src/render/CodaProcess.re (hunk 1 above)
Comment (figitaki) [reply frontend/wallet/src/render/CodaProcess.re]: Take a look now, changed to use defaultPeers.

File: frontend/wallet/src/render/views/onboarding/RunNodeStep.re (hunk 5 above)
Comment (figitaki) [reply frontend/wallet/src/render/views/onboarding/RunNodeStep.re]: Fixed.

**Comment by figitaki** [comment]
Opened a follow on with rest of the Custom Path: https://github.com/CodaProtocol/coda/pull/4625

**Pull request merged**